src/bodywork/k8s/secrets.py

```diff
@@ -162,25 +189,32 @@ def delete_secret(namespace: str, name: str) -> None:
     k8s.CoreV1Api().delete_namespaced_secret(namespace=namespace, name=name)
 
 
-def list_secrets(namespace: str, group: str = None) -> Dict[str, Dict[str, str]]:
+def list_secrets(namespace: str, group: str = None) -> Dict[str, Secret]:
```

Comment:
Should be `group: Optional[str] = None` - I'm amazed that MyPy didn't complain about this. I hope I haven't badly configured the settings...

Reply:
Good spot, that's weird it didn't get picked up. I had a quick look and our MyPy is using the version that should check this as this was allowed in older versions of the PEP rules.

Also MyPy looks configured correctly, my IDE was picking this up as optional without the explicit declaration as well.

Reply:
Okay, so you can now have implicit optional types - see [here](https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling).

I don't mind either way, but we should keep it uniform across the codebase. This might mean using `no_implicit_optional` in the MyPy config file?

Reply:
My vote is for the implied version as it is less verbose and more natural for me as that is similar to how they are defined in C#. 

I agree they should be the same everywhere. I don't mind doing the change with a replace all but I would create a separate PR for doing this to keep things clean.

Reply:
I agree with you - go for implied and then change them all in one dedicated PR.

Reply:
Cool, i'll do this once all the outstanding PR's have been merged.